extensions/sql-database-projects/src/test/projectTree.test.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as should from 'should';
+import * as vscode from 'vscode';
+
+import { Project } from '../models/project';
+import { FolderNode, FileNode, sortFileFolderNodes } from '../models/tree/fileFolderTreeItem';
+import { ProjectRootTreeItem } from '../models/tree/projectTreeItem';
+
+describe('Project Tree tests', function (): void {
+	it('Should correctly order tree nodes by type, then by name', async function (): Promise<void> {
```

Comment:
Can we have one more test with deeper file structure, to verify the functionality works as expected?

Reply:
Since VSCode requests the children of a node per-node rather than all at once, only one layer of children can be sorted at a time.  I'll make a task so I remember to add tests for a deep tree itself.

Reply:
Sounds good.
I have one question here for my own understanding: the ordering here. This tree ordering is for the opened project in Explorer, right? So, what is the behavior right now for- SchemaObjectType, where the tree would be deeper than one level?

Reply:
Actually, I kinda killed two birds with one stone with the new test case, since it also confirms ordering at all depths!